subsystem/ams_subscriptions_test.go

```diff
@@ -0,0 +1,193 @@
+			err = wiremock.createStubsForCreatingAMSSubscription(http.StatusOK)
+			Expect(err).ToNot(HaveOccurred())
+		})
+
+		// ATTENTION: this test override a wiremock stub - DO NOT RUN IN PARALLEL
+		It("UpdateSubscription failed", func() {
+
+			// create subscription
+			clusterID, err := registerCluster(ctx, userBMClient, "test-ams-subscriptions-cluster", pullSecret)
+			Expect(err).ToNot(HaveOccurred())
+			log.Infof("Register cluster %s", clusterID)
+
+			// update subscription with openshfit (external) cluster ID
+			registerHostsAndSetRoles(clusterID, 3)
+			c := installCluster(clusterID)
+			for _, h := range c.Hosts {
+				updateProgress(*h.ID, clusterID, models.HostStageDone)
+			}
+			waitForClusterState(ctx, clusterID, models.ClusterStatusFinalizing, defaultWaitForClusterStateTimeout, clusterFinalizingStateInfo)
+
+			err = wiremock.createStubsForUpdatingAMSSubscription(http.StatusUnauthorized)
+			Expect(err).ToNot(HaveOccurred())
+
+			err = completeInstallation(agentBMClient, clusterID)
+			Expect(err).To(HaveOccurred())
+
+			err = wiremock.createStubsForUpdatingAMSSubscription(http.StatusServiceUnavailable)
+			Expect(err).ToNot(HaveOccurred())
+
+			err = completeInstallation(agentBMClient, clusterID)
+			Expect(err).To(HaveOccurred())
+
+			// override back to keep other tests consistent tests
+			err = wiremock.createStubsForUpdatingAMSSubscription(http.StatusOK)
+			Expect(err).ToNot(HaveOccurred())
+		})
+	})
+
+	Context("AMS subscription on cluster deletion", func() {
+
+		It("happy flow - delete 'reserved' subscription on cluster deletion", func() {
+
+			// create subscription (in 'reserved' status)
+			clusterID, err := registerCluster(ctx, userBMClient, "test-ams-subscriptions-cluster", pullSecret)
+			Expect(err).ToNot(HaveOccurred())
+			log.Infof("Register cluster %s", clusterID)
+
+			// delete 'reserved' subscription
+			_, err = userBMClient.Installer.DeregisterCluster(ctx, &installer.DeregisterClusterParams{ClusterID: clusterID})
+			Expect(err).ToNot(HaveOccurred())
+		})
+
+		It("happy flow - don't delete 'active' subscription on cluster deletion", func() {
+
+			// create subscription
+			clusterID, err := registerCluster(ctx, userBMClient, "test-ams-subscriptions-cluster", pullSecret)
+			Expect(err).ToNot(HaveOccurred())
+			log.Infof("Register cluster %s", clusterID)
+
+			// update subscription with 'active' status
+			registerHostsAndSetRoles(clusterID, 3)
+			c := installCluster(clusterID)
+			for _, h := range c.Hosts {
+				updateProgress(*h.ID, clusterID, models.HostStageDone)
+			}
+			waitForClusterState(ctx, clusterID, models.ClusterStatusFinalizing, defaultWaitForClusterStateTimeout, clusterFinalizingStateInfo)
+			err = completeInstallation(agentBMClient, clusterID)
+			Expect(err).ToNot(HaveOccurred())
+			waitForClusterState(ctx, clusterID, models.ClusterStatusInstalled, defaultWaitForClusterStateTimeout, "installed")
+
+			// don't delete 'active' subscription
```

Comment:
How the test check that the subscription is not deleted?

Reply:
It doesn't, it just check that there were no errors in the flow, this is the best I came up with.
I will update the comment to be a little bit more clear.

Reply:
updated.